languages/rust/oso/src/host/to_polar.rs

```diff
@@ -143,3 +135,40 @@ impl<C: 'static + Clone + super::HostClass> ToPolar for C {
         })
     }
 }
+
+use std::iter;
+
+pub type PolarIter = Box<dyn Iterator<Item = Result<Box<dyn ToPolar>, crate::OsoError>>>;
+
+// Trait for the return value of class methods.
+// This allows us to return polar values, as well as options and results of polar values.
+// Iterators and futures too once we get to them.
+pub trait ToPolarIter {
+    fn to_polar_iter(&self) -> PolarIter;
```

Comment:
Should this function consume self by value? That would remove the need for clone in some implementations.

Reply:
This doesn't work because when I call it on a `dyn ToPolarIter` it cant move out of that because it doesn't know the size.